src/marketplace/PichiMarketplace.sol

```diff
@@ -0,0 +1,296 @@
+    /// @param weth_ Address of wrapped ether
+    /// @param marketplaceFeeRecipient_ Address of fee recipient
+    /// @param marketplaceFee_ Marketplace fee
+    /// @param precision_ Denominator for marketplace fee
+    function initialize(address weth_, address marketplaceFeeRecipient_, uint256 marketplaceFee_, uint256 precision_)
+        external
+        initializer
+    {
+        if (weth_ == address(0) || marketplaceFeeRecipient_ == address(0)) revert InvalidAddress();
+        if (precision_ == 0) revert InvalidValue();
+        domainSeparator = keccak256(
+            abi.encode(
+                keccak256("EIP712Domain(string name,string version,uint256 chainId, address verifyingContract)"),
+                keccak256("PichiMarketplace"),
+                keccak256(bytes("1")),
+                block.chainid,
+                address(this)
+            )
+        );
+        weth = weth_;
+        marketplaceFeeRecipient = marketplaceFeeRecipient_;
+        marketplaceFee = marketplaceFee_;
+        precision = precision_;
+
+        __Ownable_init();
+    }
+
+    /// @notice Cancels all orders for a user by setting their userMinOrderNonce to minNonce specified
+    /// @param minNonce Minimum nonce for orders that can be executed
+    function cancelAllOrdersForCaller(uint256 minNonce) external override {
+        if (minNonce <= userMinOrderNonce[msg.sender]) revert NonceLowerThanCurrent();
+
+        userMinOrderNonce[msg.sender] = minNonce;
+
+        emit AllOrdersCancelled(msg.sender, minNonce);
+    }
+
+    /// @notice Cancels specific order nonces for a user
+    /// @param orderNonces Array of order nonces to mark as cancelled
+    function cancelOrdersForCaller(uint256[] calldata orderNonces) external override {
+        if (orderNonces.length == 0) revert ArrayEmpty();
+
+        for (uint256 i = 0; i < orderNonces.length; i++) {
+            if (orderNonces[i] <= userMinOrderNonce[msg.sender]) revert NonceLowerThanCurrent();
+            if (isUserNonceExecutedOrCancelled[msg.sender][orderNonces[i]]) revert OrderAlreadyCancelled();
+            isUserNonceExecutedOrCancelled[msg.sender][orderNonces[i]] = true;
+        }
+
+        emit OrdersCancelled(msg.sender, orderNonces);
+    }
+
+    /// @notice Called by buyer to execute a seller's listing
+    /// @notice Listing must be for a collection that is accepted that is paid in an accepted currency
+    /// @param listing Listing order data
+    function executeListing(Listing calldata listing) external payable override {
+        if (!isCurrencyAccepted[listing.order.currency]) revert CurrencyNotAccepted();
+        if (!isCollectionAccepted[listing.order.collection]) revert CollectionNotAccepted();
+        if (msg.sender == listing.seller) revert OrderCreatorCannotExecute();
+
+        _validateListing(listing);
+
+        if (msg.value != 0) {
+            if (listing.order.currency != weth) revert CurrencyMismatch();
```

Comment:
Seems like weth and eth are used interchangeably. Is wrapped eth actually supported?

Reply:
The logic here is that if the buyer is purchasing the wallet with ETH, the order has to have WETH as the currency requested. The buyer can also purchase the wallet with WETH if desired, which would go through the ERC20 flow for transferring funds.